openassessment/xblock/submission_mixin.py

```diff
@@ -288,10 +327,46 @@ def create_submission(self, student_item_dict, student_sub_data):
 
         return submission
 
+    def collect_files_for_submission(self, student_sub_dict):
+        """ Collect files from CSM for individual submisisons or SharedFileUpload for team submisisons. """
+
+        if not self.file_upload_type:
+            return
+
+        for field in ('file_keys', 'files_descriptions', 'files_names', 'files_sizes'):
+            student_sub_dict[field] = []
+
+        if self.teams_enabled:
+            for team_file_upload in self.get_team_files():
+                student_sub_dict['file_keys'].append(team_file_upload.file_key)
+                student_sub_dict['files_descriptions'].append(team_file_upload.description)
+                student_sub_dict['files_names'].append(team_file_upload.item_id)
+                student_sub_dict['files_sizes'].append(team_file_upload.size)
+
+        else:
+            for file_upload in self.file_manager.get_uploads():
+                student_sub_dict['file_keys'].append(file_upload.key)
+                student_sub_dict['files_descriptions'].append(file_upload.description)
+                student_sub_dict['files_names'].append(file_upload.name)
+                student_sub_dict['files_sizes'].append(file_upload.size)
```

Comment:
What about resolving the duplication of the two blocks by creating a method like:
`def create_submission_entry(self,keys,descriptions,names.sizes):` which would set the dictionary to the data supplies depending on the block?